app/src/main/java/com/github/h3lp3rs/h3lp/AwaitHelpActivity.kt

```diff
@@ -146,19 +146,48 @@ class AwaitHelpActivity : AppCompatActivity() {
     }
 
     /**
-     *  Opens the phone call app with the emergency number from the country the
-     *  user is currently in dialed. Called either from the popup or the button.
+     *  Called when the user presses the emergency call button. Opens a pop-up
+     *  asking the user to choose whether they want to call local emergency
+     *  services or their emergency contact, and dials the correct number.
      */
+    @RequiresApi(Build.VERSION_CODES.O)
     fun emergencyCall(view: View) {
-        val emergencyNumber =
-            LocalEmergencyCaller.getLocalEmergencyNumber(
-                currentLong,
-                currentLat,
-                this
-            )
-
-        val dial = "tel:$emergencyNumber"
-        startActivity(Intent(Intent.ACTION_DIAL, Uri.parse(dial)))
+        val builder = AlertDialog.Builder(this)
+        val emergencyCallPopup = layoutInflater.inflate(R.layout.emergency_call_options, null)
+
+        builder.setCancelable(false)
+        builder.setView(emergencyCallPopup)
+
+        val alertDialog = builder.create()
+
+        // ambulance button
+        emergencyCallPopup.findViewById<ImageButton>(R.id.ambulance_call_button).setOnClickListener {
+            // In case the getCurrentLocation failed (for example if the location services aren't
+            // activated, currentLocation is still null and the returned phone number will be the
+            // default emergency phone number
+            val emergencyNumber =
+                LocalEmergencyCaller.getLocalEmergencyNumber(
+                    currentLong,
+                    currentLat,
+                    this
+                )
+
+            val dial = "tel:$emergencyNumber"
+            alertDialog.cancel()
+            startActivity(Intent(Intent.ACTION_DIAL, Uri.parse(dial)))
+        }
+
+        // contact button
+        emergencyCallPopup.findViewById<ImageButton>(R.id.contact_call_button).setOnClickListener {
+            val medicalInfo = LocalStorage(getString(R.string.medical_info_prefs),this,false)
```

Comment:
IMPORTANT: You should use storageOf, else you break the abstraction and this can be very dangerous!

Indeed, the class could not have been made protected for some odd reasons, we should look into that so that we never do this mistake again.

Reply:
It's also cleaner